### DISCUSSION TITLE
Bin numarasiyla ile taksit sorgulama

### DISCUSSION THREAD

**Discussion by nuryagdym** [mewebstudio/pos#248] [General]
Kutuphaneye BIN numarasiyla ile taksit sorgulamayi eklemeyi dusunuyorum.
Elimdeki gateway dokumanlarin 2 tanesinde BIN numarasiyla ile taksit sorgulama ornegi bulunmakta. 

Gelistirmeye baslamadan once en az 5 gateway'in BIN numarasiyla ile taksit sorgulama dokumantasyonunu olmasini istiyorum.

Elinizde bununla ilgili dokumantasyon ve ornek kodlar varsa paylasminizi rica ederim.

**Comment by husmen73** [comment]
6 Haziran 2024'te bana gelen BIN listeleri.


> YapıKredi bankasından gelen BIN listesi.
> 
> [8_digit_bin_listesi_yapikredi_gunel_new1.xlsx](https://github.com/user-attachments/files/15747473/8_digit_bin_listesi_yapikredi_gunel_new1.xlsx)

> Garantiden gelen BIN listesi (diğer bankalarda var.)
> 
> [BIN_Listesi_03062024_142403.xls](https://github.com/user-attachments/files/15747474/BIN_Listesi_03062024_142403.xls)

**Comment by proff56** [comment]
bın listesi hafıza için json dosyası veya sqlite hafif db ile kayıt edilebilir.
kullanıcı kart bilgileri girince otomatik pos seçimi olabilir.
mutli pos kullanan firmalar da çok etkili olur.

**Comment by nuryagdym** [comment]
BIN sorgulama ile ilgili ben mi yanlis anliyorum, anlamadim.

Benim bildigim kadariyla banka API'ndan kart numarasiyla taksit secenekleri sorgulayabiliyoruz.
Bu sorguyu yapabilmek icin ornek dokumantasyonlar ve varsa ornek kodlar istedim.

BIN listeyi sorgulama yontemi varsa onu da ekleyebilirim kutuphaneye.
Ama listeyi kutuphane icinde tutmak gibi bir durum olmaz.

**Comment by proff56** [comment]
gsm numaralarında olduğu gibi mesela 0532 türkcell 0542 vodafone vb. 
her operatörün numara bloğu var.  aynı şekilde de kart numaralara blokları bankalara dağıtılmış.
  alt numaraları kendi üretiyor.
  
  mesela kart numarası 492193  ile başlıyorsa  Kartı sağlayan kurum "T. GARANTİ BANKASI   A.Ş."
  alt yapı  ise "VISA"dır
  
  yine kart numarasını "97926500" ile başlayan kullanıcının kartı "TROY karrtır." ve "Debit Kart" dır. yani taksit yapılamaz.
  
  bin listesi 
  kartın ait olduğu banka ve kart türü bilgisini verir ve merkezi bir datadır.
  
  kullanın girdiği kart bilgileri  Fetch ile örnk  x.com/binApi.json adresinden kontrol edilecek
   ugunluk durumuna göre gerekli taksit isteği atılabilir.
   mesela kart "Debit Kart" ise taksit çıkmaması lazım 
   
   
   ayrıca örnek olarak 492193   numaralı kart  "T. GARANTİ BANKASI   A.Ş." ait se 
   
   zaten vakif bankın sanal posunda taksit olmayabilir 
   yani her banka kendi  kartlarına taksit yapabilir 
   ama bunu güncel bilgiler ile teyit etmek lazım
   
   
   diyelim ki aynı anda 3 pos aktif
   kullanın girdiği kart hangi bankaya ait ise 3 postan birisi ile eşleşiyorsa 
    ilgi posa yönlendilirse daha az komisyon öder.
    ayrıca taksit imkanı sağlar.
    
https://prnt.sc/EncHg30_AeFZ
